### PR TITLE
feat(bug-report): improve inline log summary and wire BlossomUploadService provider

### PR DESCRIPTION
**Pull request by mbradley**
## Summary

**Support Center fixes:** Restores the Feature Request option (was missing after the drawer replacement), relabels and reorders the support options, and removes a duplicate `feat:` prefix that was being applied to feature request subjects. Bug/feat categorization is handled in the Zendesk-to-GitHub webhook, not the client.

**Bug report log summary:** The app was sending only the first 50 log entries with bug reports -- all startup initialization noise. This rewrites the inline summary to prioritize the last 200 errors/warnings plus the last 50 entries of any level (deduped, chronological). Compare #2582 (before, startup noise) with #2628 (after, useful diagnostic context).

Also wires `BlossomUploadService` into the `bugReportServiceProvider`, which was missing (the service accepts it but the provider never passed it). This is needed for the existing NIP-17 DM bug report path.

## What was considered and pulled back

The existing `sendBugReportToRecipient` code path uploads full logs to Blossom and sends the URL via encrypted NIP-17 DM. We wired that into the Zendesk path too, but removed it after testing (51cb84f). Blossom is public storage and the URL would end up in GitHub issues via Zendesk, exposing behavioral timing and interaction patterns beyond what's already public on Nostr. The `uploadFullLogs()` method remains available for the NIP-17 DM path but is intentionally not called from the Zendesk flow.

Agree that where we draw the line on user diagnostic data in public issue flows is a broader question, thanks @NotThatKindOfDrLiz!

## Files changed

- `bug_report_dialog.dart` -- extracted `buildLogsSummary()` with error/warning prioritization
- `zendesk_support_service.dart` -- updated log section header
- `bug_report_service.dart` -- `uploadFullLogs()` method (used by NIP-17 path, not Zendesk)
- `app_providers.dart` -- wire `BlossomUploadService` into provider
- `feature_request_dialog.dart` -- remove duplicate feat: prefix and stale comment
- `support_center_page.dart` -- restore Feature Request option, relabel and reorder

## Test plan

- [x] Unit tests for `uploadFullLogs()` (success, failure, null service)
- [x] Unit tests for `buildLogsSummary()` (prioritization, dedup, ordering, edge cases)
- [x] Existing dialog widget tests pass
- [x] Manual: submitted bug report from simulator, verified improved logs in Zendesk ticket and GitHub issue (#2628)
- [x] flutter analyze clean
- [x] 6,693 tests passing in CI (1 unrelated failure in `profile_header_widget_test.dart`)

Relates to #2636

### CODE REVIEW

File: mobile/lib/widgets/feature_request_dialog.dart [hunk 1]
```diff
@@ -60,7 +60,7 @@ class _FeatureRequestDialogState extends State<FeatureRequestDialog> {
     try {
       // Submit feature request to Zendesk
       // Prefix subject with "feat:" for ticket categorization
```
Comment (hm21) [inline mobile/lib/widgets/feature_request_dialog.dart]: Nit: I think we can remove that comment line since we no longer use "feat:" as prefix

File: mobile/lib/services/bug_report_service.dart [hunk 2]
```diff
@@ -161,6 +161,49 @@ class BugReportService {
...
+  /// Returns the Blossom URL on success, null on any failure.
+  /// Best-effort: failures are logged and return null, never throws.
+  Future<String?> uploadFullLogs(BugReportData data) async {
+    if (_blossomUploadService == null) {
+      Log.debug(
+        'BlossomUploadService not available, skipping full log upload',
+        category: LogCategory.system,
+      );
+      return null;
+    }
```
Comment (hm21) [inline mobile/lib/services/bug_report_service.dart:174]: I saw that this method isn’t called anywhere in the code. I guess we just need to call it above in `collectDiagnostics` to include the URL from this method, right?
